### DISCUSSION TITLE
Comparison "fixed h method" (opty 1.12) vs. "variable h method"

### DISCUSSION THREAD

**Discussion by Peter230655** [csu-hmc/opty#176] [Q&A]
I found this ‘strange’ behaviour:
 
Using the “fixed h method”, I find a solution which gets the pendulum in an upright position in, say, duration = 7.5 sec, num_nodes = 250
Now, I run the ” variable h method” with exactly the same initial conditions, and I set initial_guess[-1] = 7.5 / 249.
 
It does find a solution, but duration = 249 * solution[-1] = 25 sec !
Intuitively one would think, it should minimally find a solution with 249 * solution[-1] <= 7.5 sec ?
 
Where is my thinking wrong?
**Thanks!**

I will be happy to share my simulation if this is of use.

**Comment by moorepants** [comment]
The amount of time it takes IPOPT to find a solution depends on many factors. It does not surprise me that it takes longer with minor changes (visible to you). You can make very minor changes in a problem and IPOPT may not even converge at all!

**Comment by Peter230655** [comment]
I think, you once told me, that optimization is a combination of math and art!

**Comment by moorepants** [comment]
I think I misunderstood this question when I answered it. I thought you were talking about how long Ipopt takes to execute but I think you are talking about the duration of the simulation solution.

**Reply by Peter230655** [reply]
Correct, I was talking how long it takes to get the pendulum to be upright.
